File: frontend/src/types/parcela.ts
// frontend/src/types/parcela.ts

export enum StatusParcela {
  Pendente = "Pendente",
  Paga = "Paga",
  Atrasada = "Atrasada",
  Cancelada = "Cancelada",
}

export interface ParcelaBase {
  pedido_id: string;
  numero_parcela: number;
  valor_parcela: number;
  data_vencimento: string; // ISO 8601 string
  data_pagamento?: string; // ISO 8601 string
  status?: StatusParcela;
}

export interface ParcelaCreate extends ParcelaBase {}

export interface ParcelaUpdate extends Partial<ParcelaBase> {}

export interface ParcelaRead extends ParcelaBase {
  id: string;
  tenant_id: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/parcela.py
# backend/app/schemas/parcela.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import date
from decimal import Decimal
from app.models.parcela import StatusParcela # Importar o Enum do modelo

class ParcelaBase(BaseModel):
    pedido_id: UUID
    numero_parcela: int
    valor_parcela: Decimal
    data_vencimento: date
    data_pagamento: Optional[date] = None
    status: Optional[StatusParcela] = StatusParcela.pendente

class ParcelaCreate(ParcelaBase):
    pass

class ParcelaUpdate(ParcelaBase):
    pass

class ParcelaRead(ParcelaBase):
    id: UUID
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
